File: spaceship_crm/java8/spaceship_crm_java8_jsp_jpa/src/main/java/com/spaceship/crm/service/SecurityService.java
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.service;

import com.spaceship.crm.bean.CustomUserDetails;
import com.spaceship.crm.entity.User;
import com.spaceship.crm.model.UserLoginModel;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;

@Service
public class SecurityService {

    @Autowired
    private UserService userService;

    @Autowired
    private AuthenticationManager authenticationManager;

    public User doLogin(UserLoginModel user) {
        UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(user.getUsername(), user.getPassword());
        Authentication auth = authenticationManager.authenticate(token);
        if (auth.isAuthenticated()) {
            SecurityContextHolder.getContext().setAuthentication(auth);
            return ((CustomUserDetails)auth.getPrincipal()).getUser();
        }
        return null;
    }

    public void doLogout() {
        SecurityContextHolder.getContext().setAuthentication(null);
    }
}

File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/service/SecurityService.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service
import com.spaceship.crm.bean.CustomUserDetails
import com.spaceship.crm.bean.UserLoginBean
import com.spaceship.crm.entity.User

@Service
class SecurityService(
        @Autowired private val userService: UserService,
        @Autowired private val authenticationManager: AuthenticationManager
) {

    fun doLogin(user: UserLoginBean): User? {
        val token = UsernamePasswordAuthenticationToken(user.username, user.password)
        val auth = authenticationManager.authenticate(token)
        if (auth.isAuthenticated) {
            SecurityContextHolder.getContext().authentication = auth
            return (auth.principal as CustomUserDetails).user
        }
        return null
    }

    fun doLogout() {
        SecurityContextHolder.getContext().authentication = null
    }


}